crates/claims/core/src/verification/mod.rs

```diff
@@ -98,3 +93,135 @@ pub enum Invalid {
     #[error("invalid proof: {0}")]
     Proof(#[from] InvalidProof),
 }
+
+/// Public key resolver environment.
+pub trait ResolverEnvironment {
```

Comment:
I don't know if environment is the right term. In SIWE for example, you provide "verification parameters" which can include a different time from now

Reply:
Yes that's the same idea with the `*Environment` traits. For now we have `DateTimeEnvironment`, `ResolverEnvironment`, `ContextLoaderEnvironment` and `Eip712TypesEnvironment`, and probably more in the future. I agree "Environment" may not be the best term, but I couldn't come up with a better one yet. Maybe "Provider".

Reply:
After talking about it with Jacob, we decided to use `Provider` instead of `Environment`